crates/ruff/src/rules/flake8_errmsg/rules.rs

```diff
@@ -163,28 +229,95 @@ pub fn string_in_exception(checker: &mut Checker, exc: &Expr) {
                 } => {
                     if checker.settings.rules.enabled(Rule::RawStringInException) {
                         if string.len() > checker.settings.flake8_errmsg.max_string_length {
-                            checker
-                                .diagnostics
-                                .push(Diagnostic::new(RawStringInException, first.range()));
+                            let indentation = whitespace::indentation(checker.locator, stmt)
+                                .and_then(|indentation| {
+                                    if checker.ctx.find_binding("msg").is_none() {
+                                        Some(indentation)
+                                    } else {
+                                        None
+                                    }
+                                });
```

Comment:
@dhruvmanila - I'm not convinced that the change I made here is _clearer_, but it does mean that we avoid the "invalid" state of `fixable = false` with `indentation = ""`.

Reply:
Ah, right. Yes, it would seem that the fixable property is dependent on whether the indentation is present or not. Maybe two variables might help but not the way I did it. 🤔

```diff
diff --git a/crates/ruff/src/rules/flake8_errmsg/rules.rs b/crates/ruff/src/rules/flake8_errmsg/rules.rs
index ac30f0d7..e828e416 100644
--- a/crates/ruff/src/rules/flake8_errmsg/rules.rs
+++ b/crates/ruff/src/rules/flake8_errmsg/rules.rs
@@ -229,20 +229,11 @@ pub fn string_in_exception(checker: &mut Checker, stmt: &Stmt, exc: &Expr) {
                 } => {
                     if checker.settings.rules.enabled(Rule::RawStringInException) {
                         if string.len() > checker.settings.flake8_errmsg.max_string_length {
-                            let indentation = whitespace::indentation(checker.locator, stmt)
-                                .and_then(|indentation| {
-                                    if checker.ctx.find_binding("msg").is_none() {
-                                        Some(indentation)
-                                    } else {
-                                        None
-                                    }
-                                });
-                            let mut diagnostic = Diagnostic::new(
-                                RawStringInException {
-                                    fixable: indentation.is_some(),
-                                },
-                                first.range(),
-                            );
+                            let indentation = whitespace::indentation(checker.locator, stmt);
+                            let fixable = indentation
+                                .map_or(false, |_| checker.ctx.find_binding("msg").is_none());
+                            let mut diagnostic =
+                                Diagnostic::new(RawStringInException { fixable }, first.range());
                             if let Some(indentation) = indentation {
                                 if checker.patch(diagnostic.kind.rule()) {
                                     diagnostic.set_fix(generate_fix(
```